nilearn/maskers/base_masker.py

```diff
@@ -277,33 +277,27 @@ def fit_transform(self, X, y=None, confounds=None, sample_mask=None,
             Transformed array.
 
         """
-        # non-optimized default implementation; override when a better
-        # method is possible for a given clustering algorithm
         if y is None:
             # fit method of arity 1 (unsupervised transformation)
-            if self.mask_img is None:
-                return self.fit(X, **fit_params
-                                ).transform(X, confounds=confounds,
-                                            sample_mask=sample_mask)
-            else:
-                return self.fit(**fit_params).transform(X,
-                                                        confounds=confounds,
-                                                        sample_mask=sample_mask
-                                                        )
-        else:
-            # fit method of arity 2 (supervised transformation)
-            if self.mask_img is None:
-                return self.fit(X, y, **fit_params
-                                ).transform(X, confounds=confounds,
-                                            sample_mask=sample_mask)
-            else:
-                warnings.warn('[%s.fit] Generation of a mask has been'
-                              ' requested (y != None) while a mask has'
-                              ' been provided at masker creation. Given mask'
-                              ' will be used.' % self.__class__.__name__)
-                return self.fit(**fit_params).transform(X, confounds=confounds,
-                                                        sample_mask=sample_mask
-                                                        )
+            return (
```

Comment:
This is the only change that I do not find great. I think it makes the code a bit harder to understand, but this may be a matter of personal taste.
The other suggestions are great imho.

Reply:
agreed that this one is a bit of an aggressive refactoring

don't know the code base enough to feel confident merging this yet

Reply:
Actually this is touching a file that is not yet part of the reformatting process.

Another reason to not touch it.

Reply:
This file is actually not yet part of the reformatting scope.

Another reason to leave it alone.